include/xgboost/json_io.h

```diff
@@ -84,8 +85,10 @@ class JsonReader {
     std::string msg = "Expecting: \"";
     msg += c;
     msg += "\", got: \"";
-    if (got == -1) {
+    if (got == EOF) {
```

Comment:
Please don't remove `-1` case.

Reply:
No, on some systems, EOF is not necessarily -1.

Reply:
Sorry, you are right. We are reading from a char array, so any EOF that appears would necessarily be -1. On ARM chips, EOF is actually bigger than char, so this case won’t apply.